runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/BossLoggerPlugin.java

```diff
@@ -0,0 +1,1365 @@
+				return;
+			}
+		}
+
+		// Raids KC
+		if (chatMessage.startsWith("Your completed Theatre of Blood count is:"))
+		{
+			Matcher m = NUMBER_PATTERN.matcher(Text.removeTags(chatMessage));
+			if (m.find())
+			{
+				killcountMap.put("RAIDS 2", Integer.valueOf(m.group()));
+				return;
+			}
+		}
+
+		// Handle all other boss
+		Matcher boss = BOSS_NAME_PATTERN.matcher(Text.removeTags(chatMessage));
+		if (boss.find())
+		{
+			String bossName = boss.group(1);
+			Matcher m = NUMBER_PATTERN.matcher(Text.removeTags(chatMessage));
+			if (!m.find())
+				return;
+			int KC = Integer.valueOf(m.group());
+			killcountMap.put(bossName.toUpperCase(), KC);
+		}
+
+		// Pet Drop
+		Matcher pet1 = PET_RECEIVED_PATTERN.matcher(Text.removeTags(chatMessage));
+		Matcher pet2 = PET_RECEIVED_INVENTORY_PATTERN.matcher(Text.removeTags(chatMessage));
+		if (pet1.find() || pet2.find())
+		{
+			gotPet = true;
+		}
+	}
+
+	private void resetStoredData()
+	{
+		for (Tab tab : Tab.values())
+		{
+			lootMap.put(tab.getBossName().toUpperCase(), new ArrayList<>());
+		}
+	}
+
+	private void updateTabData()
+	{
+		// Do nothing is panel isn't enabled
+		if (!bossLoggerConfig.showLootTotals())
+		{
+			return;
+		}
+
+		for (Tab tab : Tab.values())
+		{
+			// Only update tabs if the tabs are being shown.
+			if (isBeingRecorded(tab.getName()))
+			{
+				panel.updateTab(tab.getName());
+			}
+		}
+	}
+
+	private void init()
+	{
+		// Create maps for easy management of certain features
+		Map<String, Boolean> mapRecording = new HashMap<>();
+		Map<String, ArrayList<LootEntry>> mapLoot = new HashMap<>();
+		Map<String, Integer> mapKillcount = new HashMap<>();
+		Map<String, String> mapFilename = new HashMap<>();
+		for (Tab tab : Tab.values())
+		{
+			String bossName = tab.getBossName().toUpperCase();
+			// Is Boss being recorded?
+			mapRecording.put(bossName, isBeingRecorded(tab.getName()));
+			// Loot Entries by Tab Name
+			ArrayList<LootEntry> array = new ArrayList<LootEntry>();
+			mapLoot.put(bossName, array);
+			// Kill Count
+			int killcount = 0;
+			mapKillcount.put(bossName, killcount);
+			// Filenames. Removes all spaces, periods, and apostrophes
+			String filename = tab.getName().replaceAll("( |'|\\.)", "").toLowerCase() + ".log";
+			mapFilename.put(bossName, filename);
+		}
+		recordingMap = mapRecording;
+		lootMap = mapLoot;
+		killcountMap = mapKillcount;
+		filenameMap = mapFilename;
+
+		// Ensure we are using the requested message coloring for in-game messages
+		updateMessageColor();
+	}
+
+	//
+	// Panel Functions
+	//
+
+	// Separated from startUp for toggling panel from settings
+	private void createPanel()
+	{
+		panel = new BossLoggerPanel(itemManager, this);
+		// Panel Icon (Looting Bag)
+		BufferedImage icon = null;
+		synchronized (ImageIO.class)
+		{
+			try
+			{
+				icon = ImageIO.read(getClass().getResourceAsStream("panel_icon.png"));
+			}
+			catch (IOException e)
+			{
+				log.warn("Error getting panel icon:", e);
```

Comment:
Again a warn with no other action, I'll stop commenting on these, but it's always worth considering whether the log at any given point needs the stack trace, or the message will suffice. 

It's not wrong, it's just a worthwhile consideration.